packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1049,33 +1049,36 @@ class ProjectFileInvalidator {
     assert(packagesPath != null);
 
     if (lastCompiled == null) {
+      // Initial load.
       assert(urisToMonitor.isEmpty);
       return <Uri>[];
     }
 
-    final List<Uri> invalidatedFiles = <Uri>[];
-    int scanned = 0;
     final Stopwatch stopwatch = Stopwatch()..start();
-    for (Uri uri in urisToMonitor) {
-      if ((platform.isWindows && uri.path.contains(_pubCachePathWindows))
-          || uri.path.contains(_pubCachePathLinuxAndMac)) {
-        // Don't watch pub cache directories to speed things up a little.
-        continue;
-      }
+
+    final List<Uri> urisToScan = <Uri>[
+      // Don't watch pub cache directories to speed things up a little.
+      ...urisToMonitor.where((Uri uri) => !_isInPubCache(uri)),
```

Comment:
If you make the getter `_isNotInPubCache`, you can use a tear-off here instead.

```
final List<Uri> urisToScan = <Uri>[
  ...urisToMonitor.where(_isNotInPubCache),
  fs.file(packagesPath).uri,
];
```

Reply:
Yeah, but I'd rather using positive naming (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-double-negatives-in-apis), and I don't see the tear-off as being a huge improvement (although I suppose it would avoid an extra function call).

I'm willing to change it if you have a strong preference.

Reply:
I'm not sure that the rule applies to methods. Regardless of how you structure it, it should read as `where is not in pub cache`, but the tearoff is more concise and avoids the `!`

Reply:
I think that guideline applies to naming things in general (at least when there aren't separate positive/negative versions), but since this is a private function and is highly localized, I don't feel strongly about it.  Changed.